lib/core/versions/latest/Encoder.ts

```diff
@@ -45,13 +53,22 @@ export default class Encoder {
       throw new SidetreeError(ErrorCode.EncoderValidateBase64UrlStringInputNotString, `Input '${input}' not a string.`);
     }
 
+    const isBase64UrlString = Encoder.isBase64UrlString(input);
+    if (!isBase64UrlString) {
+      throw new SidetreeError(ErrorCode.EncoderValidateBase64UrlStringInputNotBase64UrlString, `Input '${input}' not a Base64URL string.`);
+    }
+  }
+
+  /**
+   * Tests if the given string is a Base64URL string.
+   */
+  public static isBase64UrlString (input: string): boolean {
```

Comment:
Why not just have a validation function that throws? I feel like that is how we've been handling these things in other classes.

Reply:
You are right there is already `validateBase64UrlString()` right above it which throws a generic error. I wanted in this case to throw a more specific error in `DocumenComposer` hence adding a generic method for this. I can remove it you feel like it is not adding much value.

Reply:
I'm indifferent, just want to make sure we follow the same pattern as much as possible throughout the codebase, no need to change if there's a reason for it.